packages/block-library/src/latest-posts/index.php

```diff
@@ -54,7 +55,12 @@ function render_block_core_latest_posts( $attributes ) {
 		$args['author'] = $attributes['selectedAuthor'];
 	}
 
-	$recent_posts = get_posts( $args );
+	$query        = new WP_Query;
+	$recent_posts = $query->query( $args );
+
+	if ( isset( $attributes['displayFeaturedImage'] ) && $attributes['displayFeaturedImage'] ) {
```

Comment:
```suggestion
	if ( ! empty( $attributes['displayFeaturedImage'] ) ) {
```
This can be simplified to a single `! empty()` call.

Reply:
This is feedback from @hellofromtonya - https://github.com/WordPress/gutenberg/pull/40571#discussion_r858859870

Reply:
@hellofromtonya, what are your thoughts on this? Is there a case where `! empty()` would return undesired results?

Reply:
Technically, yes, both patterns result in the same boolean state https://3v4l.org/q37iv. The design used made sure the key existed before the truthy check, whereas `empty()` hides away the intent of checking that the key actually exists.

But there's a missed opportunity here: What should the data type be for the `'displayFeaturedImage'` attribute? I'm assuming it's a true or false / turn it on or off state and not any non-falsey value. Right? 

What do I mean? I'm assuming this attribute expects a true or false to mean "yes turn this on" or "no turn it off". So it would not expect a non-empty array or string value. Rather, it would expect a boolean or integer/string representation of a boolean. It should not expect any truthy or falsey value that PHP juggles into a boolean.

IMO This conditional needs to:
- ensure the attribute key exists
- its value is a `true` state: boolean `true`, string `'1'`, or integer `1`

What value and data type(s) will be in this specific attribute?

Reply:
I personally agree with Tonya on this one. I think it makes the code more readable and clear with the isset.

Reply:
That value should always be boolean and default to false. 

https://github.com/WordPress/gutenberg/blob/033a06f731c333f2c3fe066cc01c543614c34c30/packages/block-library/src/latest-posts/block.json#L60-L63

Reply:
If the attributes can be filtered, the value and data type could change. If that's the case, then my preference is this pattern:

```suggestion
isset( $attributes['displayFeaturedImage'] ) && true === $attributes['displayFeaturedImage']
```

Else, if the attributes are not filterable, then either `isset() && $the_variable` or `! empty()` patterns work the same. Then it's preference and one that should be documented in the handbook for consistency.

Reply:
That said, both patterns exist in Core and Gutenberg. IMO the discussion of which pattern to use likely belongs in a separate coding standard discussion / issue. Once there's consensus, then the pattern used here can be changed without effect or BC concerns.

Reply:
Agreed Tonya, about both the pattern discussion and the preferred pattern if the attributes can be filtered.